mindsdb/interfaces/knowledge_base/controller.py

```diff
@@ -285,9 +292,9 @@ def _adapt_column_names(self, df: pd.DataFrame) -> pd.DataFrame:
             # wrong name
             id_column = None
 
-        if id_column is None and TableField.ID.value in columns:
-            # default value
-            id_column = TableField.ID.value
+        # if id_column is None and TableField.ID.value in columns:
+        #     # default value
+        #     id_column = TableField.ID.value
```

Comment:
This removes logic to use default 'id' column if `id_column` is not defined.

Swapping these two blocks should fix the issue:
First sets default value for id, second clears column id if setting is set but column is not exists
```python
        if id_column is None and TableField.ID.value in columns:
            # default value
            id_column = TableField.ID.value

        if id_column is not None and id_column not in columns:
            # wrong name
            id_column = None

```

Reply:
for second row, I think it should just be 2 and not 2_content, for first row it looks correct as we specify 'index' is the id_column. so it treats 'id' column as a content column

Reply:
**Description:**

1. **Multiple Content Columns:**
   - Current behavior: When multiple content columns (e.g., `content` and `content2`) are specified in an `INSERT` statement, separate records are created with IDs like `{id}_content` and `{id}_content2`.
   - Question: Should the content from these columns be appended into a single document (with a single ID) or remain as separate documents?

   **Example:**
   ```sql
   INSERT INTO world_news_kb (
       SELECT 1 AS index, 'v1' AS content, 'v2' AS content2 
       FROM example_db.home_rentals LIMIT 1
   );
   ```
   - Current Result:
     - ID: `1_content`, Content: `v1`
     - ID: `1_content2`, Content: `v2`
   - Expected Result: ???

2. **Chunked Content:**
   - Current behavior: When content is too long and split into chunks, all chunks are assigned the same ID, which causes duplicates. During deduplication, only the first chunk is kept.
   - Question: How should IDs for chunks be handled? Options:
     - Use a suffix like `{id}_chunk_{1, 2, 3}` for each chunk.
     - Other suggestions?

   **Example:**
   ```sql
   INSERT INTO world_news_kb (
       SELECT 2 AS index, 'very_long_text_will_divided_in_chunks' AS content 
       FROM example_db.home_rentals LIMIT 1
   );
   ```
   - Current Result:
     - All chunks have ID: `2`
   - Expected Result: ???